panther_hardware_interfaces/src/panther_system.cpp

```diff
@@ -0,0 +1,516 @@
+// Copyright 2023 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <panther_hardware_interfaces/panther_system.hpp>
+
+#include <rclcpp/logging.hpp>
+
+#include <hardware_interface/types/hardware_interface_type_values.hpp>
+
+#include <panther_hardware_interfaces/utils.hpp>
+
+// todo: add user variable to the script that will trigger DOUT4 instead of safety stop
+
+namespace panther_hardware_interfaces
+{
+
+using namespace std::literals;
+
+void PantherSystem::CheckJointSize()
+{
+  if (info_.joints.size() != kJointsSize) {
+    throw std::runtime_error(
+      "Wrong number of joints defined: "s + std::to_string(info_.joints.size()) + ", "s +
+      std::to_string(kJointsSize) + "expected."s);
+  }
+}
+
+void PantherSystem::SortJointNames()
+{
+  // Sort joints names - later hw_states and hw_commands are accessed by static indexes, so it
+  // is necessary to make sure that joints are in specific order and order of definitions in URDF
+  // doesn't matter
+  for (std::size_t i = 0; i < kJointsSize; i++) {
+    for (std::size_t j = 0; j < kJointsSize; j++) {
+      if (info_.joints[j].name.find(joint_order_[i]) != std::string::npos) {
+        joints_names_sorted_[i] = info_.joints[j].name;
+      }
+    }
+  }
+}
+
```

Comment:
Please add comment motivating the brutforce approach

Reply:
There are not many joints, so I think that there is no point in trying to optimize it. For loops used simply because it is more readable (I was considering some more fancy methods, but it looked bad). I think that it isn't that important information, but if you would like, I can also add it as a comment.